nilearn/datasets/atlas.py

```diff
@@ -1185,3 +1186,107 @@ def fetch_atlas_pauli_2017(version='prob', data_dir=None, verbose=1):
+        Spatial resolution of atlas image in mm {1 (default), 2}
+
+    data_dir: string
+        directory where data should be downloaded and unpacked.
+
+    base_url: string
+        base_url of files to download (None results in default base_url).
+
+    resume: bool
+        whether to resumed download of a partly-downloaded file.
+
+    verbose: int
+        verbosity level (0 means no message).
+
+    Returns
+    -------
+    data: sklearn.datasets.base.Bunch
+        Dictionary-like object, contains:
+
+        - maps: 3D Nifti image, values are indices in the list of labels.
+        - labels: ROI labels including Yeo-network annotation,list of strings.
+        - description: A short description of the atlas and some references.
+
+    References
+    ----------
+    For more information on this dataset, see
+    https://github.com/ThomasYeoLab/CBIG/tree/v0.8.1-Schaefer2018_LocalGlobal/stable_projects/brain_parcellation/Schaefer2018_LocalGlobal
+
+    Schaefer A, Kong R, Gordon EM, Laumann TO, Zuo XN, Holmes AJ,
+    Eickhoff SB, Yeo BTT. Local-Global parcellation of the human
+    cerebral cortex from intrinsic functional connectivity MRI,
+    Cerebral Cortex, 29:3095-3114, 2018.
+
+    Yeo BT, Krienen FM, Sepulcre J, Sabuncu MR, Lashkari D, Hollinshead M,
+    Roffman JL, Smoller JW, Zollei L., Polimeni JR, Fischl B, Liu H,
+    Buckner RL. The organization of the human cerebral cortex estimated by
+    intrinsic functional connectivity. J Neurophysiol 106(3):1125-65, 2011.
+
+    Licence: MIT.
+    """
+    # check inputs
+    valid_n_rois = [100, 200, 300, 400, 500, 600, 800, 1000]
+    valid_yeo_networks = [7, 17]
+    valid_resolution_mm = [1, 2]
+    if n_rois not in valid_n_rois:
+        raise ValueError("Requested n_rois={} not available. Valid "
+                         "options: {}".format(n_rois, valid_n_rois))
+    if yeo_networks not in valid_yeo_networks:
+        raise ValueError("Requested yeo_networks={} not available. Valid "
+                         "options: {}".format(yeo_networks,valid_yeo_networks))
+    if resolution_mm not in valid_resolution_mm:
+        raise ValueError("Requested resolution_mm={} not available. Valid "
+                         "options: {}".format(resolution_mm, valid_resolution_mm))
+
+    if base_url is None:
+        base_url = ('https://raw.githubusercontent.com/ThomasYeoLab/CBIG/'
+                    'v0.8.1-Schaefer2018_LocalGlobal/stable_projects/'
+                    'brain_parcellation/Schaefer2018_LocalGlobal/'
+                    'Parcellations/MNI/'
+                    )
+
+    opts = {}
+    files = []
+    labels_file_template = 'Schaefer2018_{}Parcels_{}Networks_order.txt'
+    img_file_template = 'Schaefer2018_{}Parcels_{}Networks_order_FSLMNI152_{}mm.nii.gz'
+    for f in [labels_file_template.format(n_rois, yeo_networks),
+              img_file_template.format(n_rois, yeo_networks, resolution_mm)]:
+        files.append((f, base_url + f, opts))
```

Comment:
What is opts used for? I dont get it atm.

Reply:
cleaned that up